File: packages/screen-brightness/android/src/main/java/io/capawesome/capacitorjs/plugins/screenbrightness/classes/options/SetBrightnessOptions.java
package io.capawesome.capacitorjs.plugins.screenbrightness.classes.options;

import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.screenbrightness.classes.CustomExceptions;

public class SetBrightnessOptions {

    private final float brightness;

    public SetBrightnessOptions(PluginCall call) throws Exception {
        this.brightness = SetBrightnessOptions.getBrightnessFromCall(call);
    }

    public float getBrightness() {
        return brightness;
    }

    private static float getBrightnessFromCall(PluginCall call) throws Exception {
        Double brightness = call.getDouble("brightness");
        if (brightness == null) {
            throw CustomExceptions.BRIGHTNESS_MISSING;
        }
        if (brightness < 0.0 || brightness > 1.0) {
            throw CustomExceptions.BRIGHTNESS_INVALID;
        }
        return brightness.floatValue();
    }
}

File: packages/screen-brightness/ios/Plugin/Classes/Options/SetBrightnessOptions.swift
import Foundation
import Capacitor

@objc public class SetBrightnessOptions: NSObject {
    let brightness: Double

    init(_ call: CAPPluginCall) throws {
        self.brightness = try SetBrightnessOptions.getBrightnessFromCall(call)
    }

    private static func getBrightnessFromCall(_ call: CAPPluginCall) throws -> Double {
        guard let brightness = call.getDouble("brightness") else {
            throw CustomError.brightnessMissing
        }
        if brightness < 0.0 || brightness > 1.0 {
            throw CustomError.brightnessInvalid
        }
        return brightness
    }
}
